sdl_android/src/main/java/com/smartdevicelink/transport/SdlBroadcastReceiver.java

```diff
@@ -222,9 +222,14 @@ public int compare(ResolveInfo resolveInfo, ResolveInfo t1) {
 									return 0;
 								}
 							});
-							String packageName = apps.get(0).activityInfo.packageName;
-							serviceIntent = new Intent();
-							serviceIntent.setComponent(new ComponentName(packageName, packageName +".SdlRouterService"));
+							do {
+								serviceIntent = new Intent();
```

Comment:
You do not need to create a new intent every iteration. You only need to overwrite the component name. This prevents us from being irresponsible with our memory management.